modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -76,116 +80,142 @@ object CurrencySnapshotCreator {
-                service
-                  .validateData(lastState, updates)
-                  .flatTap { validated =>
-                    Applicative[F].unlessA(validated.isValid)(logger.warn(s"Data application is invalid, errors: ${validated.toString}"))
-                  }
-                  .map(_.isValid)
-                  .handleErrorWith(err =>
-                    logger.error(err)(s"Unhandled exception during validating data application, assumed as invalid") >> false.pure[F]
-                  )
-                  .ifM(
-                    service
-                      .combine(lastState, updates)
-                      .flatMap { state =>
-                        service.serializeState(state)
-                      }
-                      .map(_.some)
-                      .handleErrorWith(err =>
-                        logger.error(err)(
-                          s"Unhandled exception during combine and serialize data application, fallback to last data application"
-                        ) >> maybeLastDataApplication.pure[F]
-                      ),
-                    logger.warn("Data application is not valid") >> maybeLastDataApplication.pure[F]
-                  )
-              }
-              .getOrElse(maybeLastDataApplication.pure[F])
-        }.flatSequence
-
-        (acceptanceResult, acceptedRewardTxs, snapshotInfo, stateProof) <- currencySnapshotAcceptanceManager.accept(
-          blocksForAcceptance,
-          lastContext,
-          lastActiveTips,
-          lastDeprecatedTips,
-          transactions =>
-            rewards
-              .map(_.distribute(lastArtifact, lastContext.snapshotInfo.balances, transactions, trigger))
-              .getOrElse(SortedSet.empty[RewardTransaction].pure[F])
-        )
+          currentEpochProgress = trigger match {
+            case EventTrigger => lastArtifact.epochProgress
+            case TimeTrigger  => lastArtifact.epochProgress.next
+          }
+          lastActiveTips <- lastArtifact.activeTips
+          lastDeprecatedTips = lastArtifact.tips.deprecated
 
-        (awaitingBlocks, rejectedBlocks) = acceptanceResult.notAccepted.partitionMap {
-          case (b, _: BlockAwaitReason)     => b.asRight
-          case (b, _: BlockRejectionReason) => b.asLeft
-        }
+          maybeLastDataApplication = lastArtifact.data
 
-        (deprecated, remainedActive, accepted) = getUpdatedTips(
-          lastActiveTips,
-          lastDeprecatedTips,
-          acceptanceResult,
-          currentOrdinal
-        )
+          maybeLastDataState <- (maybeLastDataApplication, maybeDataApplicationService).mapN {
+            case (lastDataApplication, (_, service)) =>
+              service
+                .deserializeState(lastDataApplication)
+                .flatTap {
+                  case Left(err) => logger.warn(err)(s"Cannot deserialize custom state")
+                  case Right(a)  => logger.info(s"Deserialized state: $a")
```

Comment:
do we want to log the whole deserialized state? especially as `info`?

Reply:
I see now it was in the code already. but just spotted this and probably we should remove this

Reply:
What should I do then? Change the log-level or remove that line completely?

Reply:
It's the whole deserialized state so I'd remove this log completly